docs/migration.md

```diff
@@ -0,0 +1,191 @@
+# Prometheus 2.0 migration guide
+
+In line with our [stability promise](https://prometheus.io/blog/2016/07/18/prometheus-1-0-released/#fine-print), the Prometheus 2.0 release contains
+a number of backwards incompatible changes.  This document aims to offer guidance on
+migrating from Prometheus 1.8 to Prometheus 2.0.
+
+## Flags
+
+The format of the Prometheus command line flags have changed.  Instead of a
+single dash, all flags now use a double dash. Common flags (`--config.file`,
+`--web.listen-address` and `--web.external-url`) are still the same but beyond
+that, almost all the storage-related flags have been removed.
+
+Some notable flags which have been removed:
+- `-alertmanager.url` In Prometheus 2.0, the command line flags for configuring
+  a static Alertmanager URL have been removed.  Alertmanager must now be
+  discovered via service discovery, see [Alertmanager service discovery](#amsd).
+
+- `-log.format` In Prometheus 2.0 logs can only be streamed to standard error.
+
+- `-query.staleness-delta` Prometheus 2.0 introduces a new mechanism for
+  handling staleness, see [Staleness](#staleness).
```

Comment:
Staleness links are now dead?

In case it would take too long to flesh a staleness section out tonight, you could gracefully degrade by linking to https://www.youtube.com/watch?v=GcTzd2CLH7I